x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/LookupFromIndexIT.java

```diff
@@ -0,0 +1,249 @@
+import org.elasticsearch.compute.operator.Driver;
+import org.elasticsearch.compute.operator.DriverContext;
+import org.elasticsearch.compute.operator.DriverRunner;
+import org.elasticsearch.compute.operator.PageConsumerOperator;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.IndexService;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.indices.breaker.NoneCircuitBreakerService;
+import org.elasticsearch.search.SearchService;
+import org.elasticsearch.search.internal.AliasFilter;
+import org.elasticsearch.search.internal.SearchContext;
+import org.elasticsearch.search.internal.ShardSearchRequest;
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.TaskId;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.xpack.core.async.AsyncExecutionId;
+import org.elasticsearch.xpack.esql.core.expression.Alias;
+import org.elasticsearch.xpack.esql.core.expression.ReferenceAttribute;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.enrich.LookupFromIndexOperator;
+import org.elasticsearch.xpack.esql.planner.EsPhysicalOperationProviders;
+import org.elasticsearch.xpack.esql.plugin.EsqlPlugin;
+import org.elasticsearch.xpack.esql.plugin.QueryPragmas;
+import org.elasticsearch.xpack.esql.plugin.TransportEsqlQueryAction;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static org.elasticsearch.test.ListMatcher.matchesList;
+import static org.elasticsearch.test.MapMatcher.assertMap;
+import static org.hamcrest.Matchers.empty;
+
+public class LookupFromIndexIT extends AbstractEsqlIntegTestCase {
+    /**
+     * Quick and dirty test for looking up data from a lookup index.
+     */
+    public void testLookupIndex() throws IOException {
+        // TODO this should *fail* if the target index isn't a lookup type index - it doesn't now.
+        int docCount = between(10, 1000);
+        List<String> expected = new ArrayList<>(docCount);
+        client().admin()
+            .indices()
+            .prepareCreate("source")
+            .setSettings(Settings.builder().put(IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 1))
+            .setMapping("data", "type=keyword")
+            .get();
+        client().admin()
+            .indices()
+            .prepareCreate("lookup")
+            .setSettings(
+                Settings.builder()
+                    .put(IndexSettings.MODE.getKey(), "lookup")
+                    // TODO lookup index mode doesn't seem to force a single shard. That'll break the lookup command.
+                    .put(IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 1)
```

Comment:
I think you need to override the `numberOfShards()` method of the test class to return 1; otherwise, a global template with more than one shards could be injected. We can discuss with Martijn the possibility of always overriding the number of shards for lookup indices.

Reply:
++